### PR TITLE
[SUPPORT] LPS-156865 Create upgrade to remove view permissions for guest users from DDMFormInstance

### PR DESCRIPTION
**Pull request by JavierMoral**
Hi team,
This pr fixes [LPS-156865](https://issues.liferay.com/browse/LPS-156865).

For context please check [this comment](https://issues.liferay.com/browse/LPS-113264?focusedCommentId=2703246&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2703246). First I created a Groovy script to fix this but I was asked to transform it into an upgrade process [see here](https://issues.liferay.com/browse/LPS-113264?focusedCommentId=2708293&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2708293).
Thanks!

### CODE REVIEW

**Comment by JavierMoral** [comment]
ci:test:relevant

**Comment by JavierMoral** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMFormInstanceUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+				while (resultSet.next()) {
+					long companyId = resultSet.getLong(1);
+
+					long primKey = resultSet.getLong(2);
+
+					_removeGuestViewPermission(companyId, primKey);
+				}
+			}
+		}
+	}
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMFormInstanceUpgradeProcess.java:90]: Hi @JavierMoral,
I think we could simplify this code by doing something like:

```
@Override
protected void doUpgrade() throws Exception {
	try (PreparedStatement selectPreparedStatement =
			connection.prepareStatement(
				"select companyId, formInstanceId from DDMFormInstance");
		ResultSet resultSet = selectPreparedStatement.executeQuery()) {

		while (resultSet.next()) {
			long companyId = resultSet.getLong(1);
			long formInstanceId = resultSet.getLong(2);

			Role guestRole = _roleLocalService.getRole(
				companyId, RoleConstants.GUEST);

			ResourcePermission resourcePermission =
				_resourcePermissionLocalService.getResourcePermission(
					companyId, _CLASS_NAME_FORM_INSTANCE,
					ResourceConstants.SCOPE_INDIVIDUAL,
					String.valueOf(formInstanceId), guestRole.getRoleId());

			if (!resourcePermission.hasActionId(ActionKeys.VIEW)) {
				continue;
			}

			_resourcePermissionLocalService.removeResourcePermission(
				companyId, _CLASS_NAME_FORM_INSTANCE,
				ResourceConstants.SCOPE_INDIVIDUAL,
				String.valueOf(formInstanceId), guestRole.getRoleId(),
				ActionKeys.VIEW);
		}
	}
}

private static final String _CLASS_NAME_FORM_INSTANCE =
	"com.liferay.dynamic.data.mapping.model.DDMFormInstance";
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by carolmariaabb** [comment]
Hi @JavierMoral, thank you for working on this.

Although I understand the purpose of creating this upgrade step, I am not convinced that this is the best solution for the described issue. There may be cases where we shouldn't remove the permissions because the admin has set them manually, for example.

Could you please open a PTR ticket so we can involve the Product Team in this discussion?

Thanks!

**Comment by carolmariaabb** [comment]
Hi @JavierMoral,

I'll close this PR for now and wait for the PTR ticket. I understand that it was suggested to create this upgrade process, but the current team would like to discuss a little more about how we can address this issue.

Thanks for your comprehension.

**Comment by carolmariaabb** [comment]
ci:close

**Comment by almonwork** [comment]
Hi @carolmariaabb, @JavierMoral ,

Provided it was agreed in https://issues.liferay.com/browse/PTR-3187 that this approach should be valid, could we move this forward?

Regards

**Comment by carolmariaabb** [comment]
Hi @JavierMoral, 

Can you please apply the changes suggested [here](https://github.com/liferay-forms/liferay-portal/pull/1788#discussion_r908432237) and resend this PR?

Thanks

**Pull request closed without merging**